src/main/java/com/google/code/or/common/util/MySQLUtils.java

```diff
@@ -134,12 +142,19 @@ public static int nanosForFractionalValue(int value, int width) {
 		return new java.sql.Timestamp(seconds * 1000L);
 	}
 
-	public static Timestamp toTimestamp2(long seconds, int fraction, int width) {
+	public static java.sql.Timestamp toTimestamp2(long seconds, int fraction, int width) {
 		final java.sql.Timestamp r = new java.sql.Timestamp(seconds * 1000L);
 		r.setNanos(nanosForFractionalValue(fraction, width));
 		return r;
 	}
 
+	public static java.sql.Timestamp toTimestamp2FromRawDatetime(long value, int fraction, int width) {
+		final long millis = toDatetime2(value, fraction, width).getTime();
+		final java.sql.Timestamp r = new java.sql.Timestamp(millis);
+		r.setNanos(nanosForFractionalValue(fraction, width));
+		return r;
+	}

```

Comment:
I haven't inspected the value (need to learn how to use the debugger in Java, could be useful), but according to the tests they are not the same thing in both cases. I can spend more time trying to understand that part.